### PR TITLE
[en] new SV agreement group

### PR DESCRIPTION
**Pull request by AzadehSafakish**
New group that handles singular subjects:

- clauses with predicate only (no subject)
- clauses with predicate + object OR prepositional phrase
- clauses with predicate + object + prepositional phrase

I'm taking a quick break right now, then coming back to write the rules for plural subjects (hopefully shouldn't take too long).
Brainstorming attached.
[brainstorming.pdf](https://github.com/languagetool-org/languagetool/files/6653187/brainstorming.pdf)

### CODE REVIEW

File: languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml [hunk 1]
```diff
@@ -19621,6 +19621,329 @@ USA
...
+                <suggestion><match no="8" postag="VBZ" /></suggestion>
+                <suggestion><match no="8" postag="VBD" /></suggestion>
+                <example correction="lists|listed">The file which was sent today <marker>list</marker> all of the topics for the meeting.</example>
+                <example correction="keeps|kept">The guy who didn't study <marker>keep</marker> staring at my notes.</example>
+                <example correction="annoys|annoyed">The woman who works here really <marker>annoy</marker> me.</example>
+                <example correction="declares|declared">That zoo that operated illegally finally <marker>declare</marker> bankruptcy.</example>
+                <example correction="loses|lost">The rookie athlete who doesn't practice <marker>lose</marker> most of his games.</example>
+                <example correction="shows|showed">The scan that is running currently <marker>show</marker> that some malware has been detected.</example>
+                <example correction="leaved|leaves|left">The guy that knows <marker>leave</marker> this morning.</example>
```
Comment (udomai) [inline languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml:19738]: To fix the test, all you need to do is fix the order of suggestions: `leaves|leaved|left` 🤓

**Comment by udomai** [comment]
That is thorough and systematic! Take your time, this rule is going to be a beauty.

File: languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml (hunk 1 above)
Comment (udomai) [reply languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml:19738]: About the "leaved" thing – at some point, we might have AI/ML help in determining whether the context allows for the plant-related sense of "to leave". "The trees have leaved to soon this spring". Might be an idea to mention during the workshop next week.

Comment (AzadehSafakish) [reply languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml:19738]: Screen Shot 2021-06-16 at 2 27 29 PM

El
Oh
El

**Comment by AzadehSafakish** [comment]
+ plural subjects
Also nixed future tense suggestions, 'will don't' isn't a very good correction.

File: languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml (hunk 1 above)
Comment (udomai) [reply languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml:19738]: Ex dee, indeed. Y "leaf", tho? The lemma is different ("leaves" should not be a VBZ of the supposed verb "leaf")!

Comment (AzadehSafakish) [reply languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml:19738]: XD

**Comment by AzadehSafakish** [comment]
1/3 of the way through the next rulegroup that handles clauses that contain subjects. Once that's finished, I'll just need to add rules that neatly handle 'to be' cases, then we can merge.

**Comment by AzadehSafakish** [comment]
I wasn't happy with the rulegroups I wrote previously, so I deleted them and did the whole thing over again... not using skips is the way to go.

Two new rule groups, one for singular and one for plural subjects.
Each group will have eight rules:

1.) subjects containing clauses with predicate only + adverbs
2.) ... clauses with a subject + predicate
3.) ... clauses with (subject - optional) + predicate + object OR prepositional phrase
4.) ... clauses with (subject - optional) + predicate + object + prepositional phrase
5-8.) handles is/are/was/were

Taking a break right now, but I'll be back to finish. I want to have this wrapped up tonight.

**Comment by AzadehSafakish** [comment]
Subrules 1-4 are complete, I want to forget about 5-8 (is/are/was/were) cases for the moment because I want to see the diff. Ready for review/merging :)

**Comment by udomai** [comment]
Manually merged: 0be1f0fb544fa9050818d09b4e45a35973e5ab74

**Pull request closed without merging**